commit 8038f92df3eb2a98d68ba78799d4026b099f6705
Author: Taehee Yoo <ap420073@gmail.com>
Date:   Sun Nov 25 18:47:13 2018 +0900

    netfilter: nf_conncount: remove wrong condition check routine
    
    [ Upstream commit 53ca0f2fec39c80ccd19e6e3f30cc8daef174b70 ]
    
    All lists that reach the tree_nodes_free() function have both zero
    counter and true dead flag. The reason for this is that lists to be
    release are selected by nf_conncount_gc_list() which already decrements
    the list counter and sets on the dead flag. Therefore, this if statement
    in tree_nodes_free() is unnecessary and wrong.
    
    Fixes: 31568ec09ea0 ("netfilter: nf_conncount: fix list_del corruption in conn_free")
    Signed-off-by: Taehee Yoo <ap420073@gmail.com>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/net/netfilter/nf_conncount.c b/net/netfilter/nf_conncount.c
index 8acae4a3e4c0..b6d0f6deea86 100644
--- a/net/netfilter/nf_conncount.c
+++ b/net/netfilter/nf_conncount.c
@@ -323,11 +323,8 @@ static void tree_nodes_free(struct rb_root *root,
 	while (gc_count) {
 		rbconn = gc_nodes[--gc_count];
 		spin_lock(&rbconn->list.list_lock);
-		if (rbconn->list.count == 0 && rbconn->list.dead == false) {
-			rbconn->list.dead = true;
-			rb_erase(&rbconn->node, root);
-			call_rcu(&rbconn->rcu_head, __tree_nodes_free);
-		}
+		rb_erase(&rbconn->node, root);
+		call_rcu(&rbconn->rcu_head, __tree_nodes_free);
 		spin_unlock(&rbconn->list.list_lock);
 	}
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1095_8038f92df3eb2a98d68ba78799d4026b099f6705.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1095_8038f92df3eb2a98d68ba78799d4026b099f6705.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
